### PR TITLE
RELATED: RAIL-3987 add support for ingoreFilters to custom widgets

### PR DESCRIPTION
**Pull request by no23reason**
Now custom widgets can choose to implement the `IFilterableWidget` interface, imitating how standard widgets specify 
* attribute filters to ignore
* date dataset to use for date filters

To leverage this, the `useWidgetFilters` was marked as public and its API made more precise.
Also, two new hooks were added to improve the DX there:
* `useCustomWidgetExecutionDataView` for freeform execution
* `useCustomWidgetInsightDataView` for inisght execution

The `IDashboardKpiProps.filters` prop was deprecated and ignored in code (it is a remnant of DashboardView).
The payload of `QueryInsightWidgetFilters` was improved for naming and the overrides only allowed for InsightWidgets (it makes very little sense for Kpi- and CustomWidgets).

`selectWidgetByRef` semantics  were changed so that it returns custom widgets as well.

The `ErrorComponent` and `LoadingComponent` of the `IDashboardWidgetProps` type were made mandatory to make them easier to use (similarly as was done in #2149 for insights).

The Dashboard example with Local plugin was extended to showcase the new hook and `IFilterableWidget` props.

There were some simplifications done in the `queryWidgetFilters` service to make the code less redundant and more to-the-point.

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/presentation/widget/kpi/types.ts [hunk 1]
```diff
@@ -55,6 +55,7 @@ export interface IDashboardKpiProps {
     /**
      * Optionally, specify filters to be applied to the KPI.
      *
+     * @deprecated Do not use, made part of the public API by accident. Will do nothing.
      * @public
      */
     filters?: FilterContextItem[];
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/widget/kpi/types.ts:61]: I wonder if we could just remove it since it was always quite useless 🤔

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6479/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8337/ : SUCCESS in 17m 07s

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6486/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8344/ : SUCCESS in 16m 44s

**Review by kandl — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2156

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1904/ : SUCCESS in 1m 05s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2185/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/988/ : SUCCESS in 15m 25s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8349/ : SUCCESS in 17m 10s

**Pull request merged**